gnucash/gnome-utils/gnc-tree-view-account.c

```diff
@@ -1881,12 +1881,25 @@ gnc_tree_view_account_add_custom_column(GncTreeViewAccount *account_view,
                                         GncTreeViewAccountColumnTextEdited
                                         col_edited_cb)
 {
-    GtkCellRenderer *renderer;
+    GtkCellRenderer *renderer = gtk_cell_renderer_text_new();
```

Comment:
In the original function it was first verified that account_view is actually of type GNC_TREE_VIEW_ACCOUNT before a renderer is created. If not the function returned NULL.

With your changes the function will always create the renderer yet it may never be used if the account_view parameter is not of type GNC_TREE_VIEW_ACCOUNT. So you're potentially leaking a renderer for each invalid call to this function.

Be sure to first test the inputs and only then create the renderer.

Reply:
fixed